pkg/registry/registry.go

```diff
@@ -203,9 +203,10 @@ func (r *PlainRegistry) parseImage(image string) (string, string, string, bool)
 		image = "library/" + image
 	}
 	for _, proxy := range r.Proxies {
-		if ok, err := filepath.Match(proxy.Registry, registry); err == nil && ok {
+		if ok, err := filepath.Match("*."+proxy.Registry, registry); err == nil && ok {
```

Comment:
I am unsure we should march registry subdomains.
I think the mentioned `something.docker.io` references `registry-1.docker.io` which is [added within noe](https://github.com/adevinta/noe/blob/89411cd31af1b048d0b9d49a16d123132999714c/pkg/registry/registry.go#L451), so [do other registry clients like containerd](https://github.com/containerd/containerd/blob/124456ef83f5984e597c4ab2b48b9074199c1aa0/core/remotes/docker/registry.go#L193)